src/Cli/dotnet/OptionForwardingExtensions.cs

```diff
@@ -18,6 +18,25 @@ public static class OptionForwardingExtensions
 
         public static ForwardedOption<T> ForwardAsSingle<T>(this ForwardedOption<T> option, Func<T, string> format) => option.SetForwardingFunction(format);
 
+        public static ForwardedOption<string> ForwardAsOutputPath(this ForwardedOption<string> option, string outputPropertyName, bool surroundWithDoubleQuotes = false)
```

Comment:
I'm not sure what you mean.  Visual Studio doesn't pass the `-o` option.  This PR is about what happens if you build a solution from the command line and specify that option.

Reply:
Sorry, I was thinking of some hypothetical implementation of VS that invoked an SDK CLI process with `-o ./project-name/`, in which case this would break that. But that doesn't happen. 

If that would be all it takes to implement the feature from the SDK side though, that would be pretty awesome